STYLE_GUIDE.md

```diff
@@ -0,0 +1,383 @@
+```
+
+**Not Preferred:**
+
+```swift
+let message: String = "Click the button"
+var currentBounds: CGRect = computeViewBounds()
+```
+
+**NOTE**: Following this guideline means picking descriptive names is even more important than before.
+
+## Syntactic sugar
+
+Prefer the shortcut versions of type declarations over the full generics syntax.
+
+**Preferred:**
+
+```swift
+var deviceModels: [String]
+var employees: [Int: String]
+var faxNumber: Int?
+```
+
+**Not Preferred:**
+
+```swift
+var deviceModels: Array<String>
+var employees: Dictionary<Int, String>
+var faxNumber: Optional<Int>
+```
+
+## Semicolons
+
+Do not use semicolons at the end of a line - they are no longer required. 
+
+Do not write multiple statements on a single line separated with semicolons.
+
+## Resources
+
+Use extensions or wrapper classes for accessing elements of asset catalogs, custom colors and fonts. It helps to avoid the error-prone practice of hard-coding strings into your code.
+
+Using generated code for this wherever possible is even better, since if an element you were previously accessing was removed, when the code is regenerated, the build will fail at compile time instead of runtime. 
+
+Use a `Theme.swift` that contains extensions for the following classes which will assist in theming your app: 
+
+- `UIColor` wrappers
+- `CGFloat` sizes for fonts and margins
+- `UIFont` wrappers
+ 
+Work with the designer on your project to come up with a common naming scheme for colors to make picking colors, fonts, and margins out of a design file easier. 
+
+### Image assets
+
+Images should live in `Images.xcassets`. They don't need to be grouped in any way unless you find that helpful for organization.
+
+Images should be named consistently within an app to preserve organization and developer sanity. It is strongly suggested to append the state name to images for non-default states.
+
+Images should be named in either `camelCase` or  `lower_snake_case` to facilitate code generation based on the image names. Pick one and stick to it per project. 
+
+**Examples:**
+
+- `ic_refresh` and `ic_refresh_selected` for a refresh icon and its selected state used in multiple places throughout an app
+- `articleNavigationBar` and `articleNavigationBarSelected` for  something used in one specific place. 
+
+### Localized strings
+
+Any strings visible to the user must be localized using `NSLocalizedString` or something wrapping it. It is way, way easier to set up all strings like this from the beginning than to add this feature later. 
+
+Don't use the developer-language content of the string as the localized string's key, because if you do, you then have to update the copy in multiple places. 
+
+Use `lower_snake_case` for localized string keys to both facilitate code generation and to make it extremely obvious when there is an untranslated user-facing string, since by default if a value does not exist for a key in `Localizable.strings`, the key itself is returned.
```

Comment:
I wrote a test helper for this years ago https://github.com/designatednerd/DNSiOSLocalizationTestHelpers - it would certainly be nice to know at compile time though